webknossos-datastore/app/com/scalableminds/webknossos/datastore/controllers/ZarrStreamingController.scala

```diff
@@ -215,75 +258,259 @@ class ZarrStreamingController @Inject()(
+                   dataLayerName,
+                   mag)
+        }
+      } yield result
   }
 
-  def zGroup(
+  def requestDataLayerMagFolderContents(token: Option[String],
+                                        organizationName: String,
+                                        dataSetName: String,
+                                        dataLayerName: String,
+                                        mag: String): Action[AnyContent] =
+    Action.async { implicit request =>
+      for {
+        result <- dataLayerMagFolderContents(urlOrHeaderToken(token, request),
+                                             organizationName,
+                                             dataSetName,
+                                             dataLayerName,
+                                             mag)
+      } yield result
+    }
+
+  private def dataLayerMagFolderContents(token: Option[String],
+                                         organizationName: String,
+                                         dataSetName: String,
+                                         dataLayerName: String,
+                                         mag: String)(implicit m: MessagesProvider): Fox[Result] =
+    accessTokenService.validateAccess(UserAccessRequest.readDataSources(DataSourceId(dataSetName, organizationName)),
+                                      token) {
+      for {
+        (_, dataLayer) <- dataSourceRepository.getDataSourceAndDataLayer(organizationName, dataSetName, dataLayerName) ~> 404
+        magParsed <- Vec3Int.fromMagLiteral(mag, allowScalar = true) ?~> Messages("dataLayer.invalidMag", mag) ~> 404
+        _ <- bool2Fox(dataLayer.containsResolution(magParsed)) ?~> Messages("dataLayer.wrongMag", dataLayerName, mag) ~> 404
+      } yield
+        Ok(
+          views.html.datastoreZarrDatasourceDir(
+            "Datastore",
+            "%s/%s/%s/%s".format(organizationName, dataSetName, dataLayerName, mag),
+            List.empty
+          )).withHeaders()
+    }
+
+  def dataLayerMagFolderContentsPrivateLink(accessToken: String,
+                                            dataLayerName: String,
+                                            mag: String): Action[AnyContent] =
+    Action.async { implicit request =>
+      for {
+        annotationSource <- remoteWebKnossosClient.getAnnotationForPrivateLink(accessToken)
+        layer = annotationSource.getAnnotationLayer(dataLayerName)
+
+        result <- layer match {
+          case Some(annotationLayer) =>
+            remoteTracingstoreClient
+              .getDataLayerMagFolderContents(annotationLayer.tracingId,
+                                             mag,
+                                             annotationSource.tracingStoreUrl,
+                                             accessToken)
+              .map(
+                layers =>
+                  Ok(
+                    views.html.datastoreZarrDatasourceDir(
+                      "Tracingstore",
```

Comment:
That seems wrong here ;) Maybe even include the info that it is this combined annotation thingy in the html page.